### PR TITLE
CI: nightly job to update latest docker tag only

### PR DESCRIPTION
**Pull request by maxknv**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
1. Nightly job do not rebuild all docker but just reset latest
2. Hard fail CI, if image tag/digest not found and docker builds are turned off
3. bug fixes: few updates in ci_config
4. removed test job DatabaseOrdinary as deprecated

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 14ff9cbf2dc49e1a97374ea7c7dc4c7fb7f60750 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59586/14ff9cbf2dc49e1a97374ea7c7dc4c7fb7f60750/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59586/14ff9cbf2dc49e1a97374ea7c7dc4c7fb7f60750/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by Felixoid — COMMENTED** [review]
Let's check if it works. I suggest updating the branch to the master and trigger the workflow from this branch https://github.com/ClickHouse/ClickHouse/actions/workflows/nightly.yml

image

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -362,14 +362,6 @@ jobs:
       test_name: Stateless tests (release)
       runner_type: func-tester
       data: ${{ needs.RunConfig.outputs.data }}
-  FunctionalStatelessTestReleaseDatabaseOrdinary:
```
[the remark this answers is not in the document]
Comment (maxknv) [reply .github/workflows/master.yml:365]: why removal is here? I asked @alesapin, he sad it could be removed. (it was disabled in ci_config)

**Comment by maxknv** [comment]
Nightly job from this branch: https://github.com/ClickHouse/ClickHouse/actions/runs/7787943923

File: tests/ci/docker_manifests_merge.py [hunk 2]
```diff
@@ -154,22 +154,28 @@ def main():
...
         if image_obj.only_amd64:
             # FIXME: WA until full arm support
             tags = [f"{tag}-{arch}" for arch in archs if arch != "aarch64"]
         else:
             tags = [f"{tag}-{arch}" for arch in archs]
-        manifest, test_result = create_manifest(image_obj.repo, tag, tags, args.push)
-        test_results.append(TestResult(manifest, test_result))
+
+        # 1. update multiarch latest manifest for every image
         if args.set_latest:
```
[the remark this answers is not in the document]
Comment (maxknv) [reply tests/ci/docker_manifests_merge.py:165]: done

File: tests/ci/ci.py [hunk 3]
```diff
@@ -950,50 +942,33 @@ def _configure_docker_jobs(
...
-            missing_aarch64 = list(
-                check_missing_images_on_dockerhub(
-                    {
-                        im: digest
-                        for im, digest in missing_multi_dict.items()
-                        if not images_info[im]["only_amd64"]
-                    },
-                    "aarch64",
-                )
+    # FIXME: we need login as docker manifest inspect goes directly to one of the *.docker.com hosts instead of "registry-mirrors" : ["http://dockerhub-proxy.dockerhub-proxy-zone:5000"]
```
[the remark this answers is not in the document]
Comment (maxknv) [reply tests/ci/ci.py:945]: done

**Comment by Felixoid** [comment]
https://github.com/ClickHouse/ClickHouse/actions/runs/7787943923/job/21236388552#step:4:114 awesome!

**Pull request merged**